File: JULY-2025/16-07-2025/p3201.go
func maximumLength(nums []int) int {
    odd,evn,alt,lst := 0,0,1,0
    if nums[0] % 2 == 1 { lst,odd = 1,1 } else { evn = 1 }
    for i:=1; i<len(nums); i++ {
        m := nums[i] % 2
        if m == 0 { evn++ } else { odd++ }
        if m != lst { alt, lst = alt+1, m }
    }
    if alt >= odd && alt >= evn { return alt }
    if odd > evn { return odd }
    return evn
}

File: JULY-2025/16-07-2025/p3201.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function maximumLength($nums) {
      $even = 0;
    $odd = 0;
    $alt = 0;
    $p = $nums[0] & 1;

    foreach ($nums as $num) {
        $bit = $num & 1;
        if ($bit == $p) {
            $alt++;
            $p = ($p == 1) ? 0 : 1;
        }
        if ($bit == 1) {
            $odd++;
        } else {
            $even++;
        }
    }

    return max($alt, max($even, $odd));  
    }
}